packages/shared/friends/sagas.ts

```diff
@@ -1179,3 +1193,241 @@ function updateSocialInfo(socialData: SocialData) {
     })
   )
 }
+
+// Join or create channel
+async function* handleJoinOrCreateChannel(action: JoinOrCreateChannel) {
+  try {
+    const client: SocialAPI | null = getSocialClient(store.getState())
+    if (!client) return
+
+    const channelId = action.payload.channelId
+    const ownId = client.getUserId()
+
+    // get or create channel
+    const { created, conversation } = await client.getOrCreateChannel(channelId, [ownId])
```

Comment:
Convert async calls this way
```suggestion
    const { created, conversation } = yield apply(client, client.getOrCreateChannel, [channelId, [ownId]])
```